### DISCUSSION TITLE
UnattendedWinstall v2.0.0

### DISCUSSION THREAD

**Discussion by memstechtips** [memstechtips/UnattendedWinstall#163] [General]
# UnattendedWinstall v2.0.0 Release

### Key Features

- Windows 10/11 Pro is installed by default
- Bypasses Windows 11 system requirements
- Disables Windows Defender services by default
  - *prompted to enable after Windows installation*
- Disables User Account Control by default
  - *prompted to enable after Windows installation*
- Allows execution of PowerShell scripts by default
- Skips forced Microsoft account creation during Windows setup
- Removes preinstalled bloatware apps except Microsoft Edge, Notepad and Calculator
  - Copilot and Recall is Disabled.
- Sets privacy-related registry keys to disable telemetry
- Limits Windows Update to install only security updates for one year
- Optimizes registry with various optimization and customization-related keys
  - *See the "Set-RecommendedHKLMRegistry" and "Set-RecommendedHKCURegistry" functions for more information*
- Disables unnecessary scheduled tasks
- Configures Windows services for optimal performance
- Enables the Ultimate Performance power plan

> [!NOTE] 
> Use the `UWScript.ps1` file once Windows is installed to reapply or revert settings in case Windows Update resets some of the settings or if you encounter any issues.  
> It can also be used to achieve a similar experience to UnattendedWinstall on an existing Windows installation without reinstalling Windows.
>
> ---
>
> **Before Running the Script**
>
> Ensure you open PowerShell as an administrator. Additionally, set the execution policy to allow script execution by running the following command:
>
> ```powershell
> Set-ExecutionPolicy RemoteSigned
> ```
>
> Running PowerShell with elevated permissions and enabling script execution will ensure that `UWScript.ps1` can apply the necessary system changes.

<hr /><em>This discussion was created from the release <a href='https://github.com/memstechtips/UnattendedWinstall/releases/tag/v2.0.0'>UnattendedWinstall v2.0.0</a>.</em>

**Comment by shimizu-izumi** [comment]
Is the autounattend.xml in this release a continuation of the IoT-LTSC-Like file?

**Reply by memstechtips** [reply]
It's kind of a mix of the previous 3 versions.
Windows Defender is Disabled by default, so in that regard it is like the Core version.
But Microsoft Edge is still installed and the only UWP apps included is the Calculator and Notepad, so in that regard it is more like the "IoT-LTSC-Like" version.

But in all honesty, it is a new file but you guys have way more options in terms of reverting or reapplying changes due to the inclusion of the `UWScript.ps1` file.

**Comment by ArakiSatoshi** [comment]
Honestly, removing presets was a mistake. The 2.0.0 version is definitely less user-friendly and confuses people who have made their own `autounattend.xml` files by removing the lines with the changes they don't need from the provided presets.

The introduction of the `UWScript.ps1` file is a nice addition on the surface, but it means that the user now has to prepare _two_ files, both `autounattend.xml` and `UWScript.ps1`, if they want to tweak it to their needs, which was the empasis of the pre-2.0.0 versions. Maybe it can be solved by pointing `UWScript.ps1` to the `autounattend.xml` file to grab the values from it instead, if that would be possible with PowerShell? Because right now people might find themselves creating the `autounattend.xml` file using the [unattend-generator](https://schneegans.de/windows/unattend-generator/) website [(1)](https://github.com/memstechtips/UnattendedWinstall?tab=readme-ov-file#automation), installing Windows [(2)](https://github.com/memstechtips/UnattendedWinstall?tab=readme-ov-file#method-1-create-a-bootable-windows-installation-usb), and then running `UWScript.ps1` [(3)](https://github.com/memstechtips/UnattendedWinstall?tab=readme-ov-file#how-can-i-apply-these-settings-to-an-existing-windows-installation) which as far as I can tell will overwrite the changes of the custom `autounattend.xml` file with the current UnattendedWinstall's values. It pretty much would be the steps to follow currently outlined in the documentation.

As a side note, I found out that the 2.0.0 is out by opening the link to the former IoT-LTSC-Like [autounattend.xml](https://github.com/memstechtips/UnattendedWinstall/blob/main/IoT-LTSC-Like/autounattend.xml) file I occasionally use for reference and finding out that the page now returns `404`. Not the best introduction to the new update...

**Reply by shimizu-izumi** [reply]
Agree and funny coincidence, I found out 2.0.0 is out the same way.

I manually applied all tweaks using PowerShell commands to my system after it automatically updated to 24H2 yesterday and used a modified version to install Windows on my sister's new laptop a few days ago.

**Reply by YaltuTheEnlightened** [reply]
I’m a tad bit bummed that there’s no longer those three different types of autoattend files we can choose from but I understand the change behind it and providing only one, but I also agree that 2.0 feels a little less user friendly as I tried 1.0 on a VM and it worked wonderfully. I’m a bit nervous to try 2.0, not only that but edge is installed by default unlike the first variation of the autoattend files provided (standard and core version). 

Is there a way to add a similar line or set of lines to the new autoattend file to prevent edge from being installed completely so that upon a fresh install of windows on my actual PC, edge is not there (and I can just use the install script packed into it to install a different browser)? It so then I’d definitely use cursor or some other text editor to add those lines in and try this new version. The problem is I don’t know what those lines are to stop edge from being installed lol or if it’s even possible with this one.

**Reply by memstechtips** [reply]
Hey everyone and thanks for the feedback, I want to make you all aware of the following regarding some of your questions here:
1. You still have access to the previous "IoT-LTSC-Like", "Standard" and "Core" versions of the file(s) here: https://github.com/memstechtips/UnattendedWinstall/releases/tag/v1.0.0
> [!NOTE]
> You need to download the `Source Code.zip` file, once extracted, you'll have access to all the previous v1.0.0 files.
2. With regards to Microsoft Edge:
- I spent a lot of time trying to properly remove it during the Windows Installation process and encountered a lot of issues mainly because of the differences between the versions in the Windows 10 22H2 ISO and Windows 11 24H2 ISO.
- I don't really want to forcibly remove Edge, I want to uninstall it using the supported method by Microsoft and I will try and incorporate it in future versions of the project.
- With that being said, if you want to forcibly remove it after Windows has been installed, I recommend using this file: https://github.com/FR33THYFR33THY/Ultimate-Windows-Optimization-Guide/blob/main/6%20Windows/14%20Edge.ps1
I got a lot of inspiration for v2.0.0 of this project from FR33THY's Ultimate Windows Optimization Guide and I recommend checking it out too.
3. With regards to the following statement:
> the user now has to prepare two files

The user does not need to prepare two files, the entire `UWScript.ps1` file is included in the `autounattend.xml` file, so the XML file is the only file that has to be edited with regards to the tweaks you want to apply during the Windows installation.
The `UWScript.ps1` file that's provided in this project is only for people who want to apply the UnattendedWinstall tweaks to an existing, stock Windows installation, without doing a clean install of Windows using the `autounattend.xml` file.
For those who install Windows using the XML file, the exact same `UWScript.ps1` file is already included in your installation and can be found in `C:\Windows\Setup\Scripts\UWScript.ps1` with a desktop shortcut to it as well.

I also understand that with UnattendedWinstall v2.0.0 it makes it much harder for people to edit the XML file because they need more advanced knowledge of PowerShell scripts specifically, but that is how this project is evolving and why I didn't completely remove the v1.0.0 files, because you can still use those to add your own tweaks.

You can also still add your own registry entries to the v2.0.0 file and it is actually easier if you understand where to add it, I'll give a brief explanation.

For registry entries that apply to the local machine, ie. `HKEY_LOCAL_MACHINE` registry keys, you can find the `function SetRecommendedHKLMRegistry` in the `autounattend.xml` file, see here: https://github.com/memstechtips/UnattendedWinstall/blob/93305192ed6d64e0f5b98a89f447927480285354/autounattend.xml#L1981

and then add whatever registry entries you want to add in `.reg` format, like the rest of the entries are set and just make sure you add it before the `"@` to make it part of the `.reg` file that will be generated, see here: https://github.com/memstechtips/UnattendedWinstall/blob/93305192ed6d64e0f5b98a89f447927480285354/autounattend.xml#L3412

and it will then be applied to the registry.

Similarly, if you have `HKEY_CURRENT_USER` registry keys, you can add those to the `User Customization.ps1` file in the same way as explained above starting here:
https://github.com/memstechtips/UnattendedWinstall/blob/93305192ed6d64e0f5b98a89f447927480285354/autounattend.xml#L3912
so below the `Windows Registry Editor Version 5.00` and then ending before the `"@` here: https://github.com/memstechtips/UnattendedWinstall/blob/93305192ed6d64e0f5b98a89f447927480285354/autounattend.xml#L4423

> [!NOTE]
> The above links might not take you to the correct lines of code once new versions of the file is released, but it does take you to the correct lines on v2.0.0

I hope this explanation has cleared up some of these issues, thanks to everyone for your interest in this project.

**Reply by YaltuTheEnlightened** [reply]
This very much cleared up a lot of my questions, thank you! I will definitely try 2.0 out now. At the end of the day edge will be there but dormant and not used (and the methods you mentioned to uninstall it), so it won't hurt anything on my pc lol. A lot of the hard work you put into these scripts is very impressive and already lightens the amount of processes running, making windows feel more lighter and snappier. I'm looking forward to the feature of this project and I hope it keeps going!

**Reply by horaceBean** [reply]
I’m not a computer technician. I ran UWScript.ps1 using PowerShell on an existing Windows 10 Pro version 22H2. The script didn’t remove Microsoft Edge (I'm okay with that), but I was disappointed that it did not remove Microsoft Store and disable unnecessary tasks. My CPU is still running 187 processes after running the UWScript.ps1 twice. And there was no prompt to restart my laptop after the script ran. I thought that was odd so I restarted the pc myself. Where did I go wrong?

**Reply by YaltuTheEnlightened** [reply]
> I’m not a computer technician. I ran UWScript.ps1 using PowerShell on an existing Windows 10 Pro version 22H2. The script didn’t remove Microsoft Edge (I'm okay with that), but I was disappointed that it did not remove Microsoft Store and disable unnecessary tasks. My CPU is still running 187 processes after running the UWScript.ps1 twice. And there was no prompt to restart my laptop after the script ran. I thought that was odd so I restarted the pc myself. Where did I go wrong?

You mention an existing windows installation. My guess with those 187 processes or so is that you _may_ have other software and stuff installed on your PC. When I was experimenting with version one, just like one of memory's past video, I was getting below 90 processes after a fresh install of the standard autoattend file. After adding games, art programs, and other applications the process count went up to 100+. There's not much you can do about that since your pc is gonna be running these processes in the foreground and background. Honestly when you get other stuff installed after a fresh install there's not much you can do with that. As for Microsoft store not uninstalling, I would just leave it be as there's no way you can remove it officially without forcibly removing it and potentially running into errors.

**Reply by memstechtips** [reply]
@horaceBean The current iteration of the `UWScript.ps1` file does not remove Microsoft Edge. It should be removing the Microsoft Store though, so I'll have to check on that and get back to you.

What @YaltuTheEnlightened explained above with regards to process count is accurate. On a new Windows installation with the UnattendedWinstall file, the process count will be less compared to a stock Windows install.
But as soon as you start installing software, that process count will start going up, because there are more programs on the computer now. The same goes for an existing installation, the script still changes some services' startup types to either "Disabled" or "Manual" but those are only a few processes.

**Reply by YaltuTheEnlightened** [reply]
A quick question now that I'm thinking about.

This may sound silly but just in case, since 2.0 has recall disabled by default, is there a command or a way to verify that recall has been disabled upon a fresh install with this new autoattend file? I could never find recall in the settings like I've seen on google or in videos lol. Thanks!

**Reply by memstechtips** [reply]
@YaltuTheEnlightened 

You can run the following commands in PowerShell as Admin to check the status.

Commands:
Check if RECALL is Enabled or Disabled:
```Dism /Online /Get-Featureinfo /Featurename:Recall```

Disable RECALL:
```Dism /Online /Disable-Feature /Featurename:Recall```

Enable RECALL:
```Dism /Online /Enable-Feature /Featurename:Recall```

**Reply by YaltuTheEnlightened** [reply]
> @YaltuTheEnlightened 
> 
> 
> 
> You can run the following commands in PowerShell as Admin to check the status.
> 
> 
> 
> Commands:
> 
> Check if RECALL is Enabled or Disabled:
> 
> ```Dism /Online /Get-Featureinfo /Featurename:Recall```
> 
> 
> 
> Disable RECALL:
> 
> ```Dism /Online /Disable-Feature /Featurename:Recall```
> 
> 
> 
> Enable RECALL:
> 
> ```Dism /Online /Enable-Feature /Featurename:Recall```

Thank you!

And one final question. So I most likely will just use the 2.0 file as is and keep it as simple as that.

Now, just in case, let's say I wanted to remove edge my self. What would I need to add into the autoattend file to make edge disappear before windows installation is complete. Do I adopt the file that you linked in your previous responses about windows optimizations and removing edge. Do I have to add a set of lines at the end of the autoattend file or does it have to be specific part of the file?

**Reply by memstechtips** [reply]
@YaltuTheEnlightened I would suggest that you use the v2.0 file as is without any modifications and then once Windows is installed, use the Edge Removal script I linked earlier.

**Reply by Plutoaurus** [reply]
I have to agree, this is almost dangerous as it disables so much by default including defender etc.

The old IoT-LTSC-Like was a good Generic user build.
Im glad ive forked before v2 and have my own custom copy.

**Reply by rautamiekka** [reply]
> I have to agree, this is almost dangerous as it disables so much by default including defender etc.

Defender is nearly as useless as having nothing, ESPECIALLY if your Internet is down, so you'll only make a mistake if you don't use anything else. In any case, you can just reactivate Defender.

**Reply by Plutoaurus** [reply]
> > I have to agree, this is almost dangerous as it disables so much by default including defender etc.
> 
> 
> 
> Defender is nearly as useless as having nothing, ESPECIALLY if your Internet is down, so you'll only make a mistake if you don't use anything else. In any case, you can just reactivate Defender.

This just isn't true in the last few years. 
Check any reviews and it scores highly and better than nothing. 

I look down the list of the most popular antivirus and there's not many I would let have full unquestioned access to my system. 
Quick research into country of origin, history, performance rules a lot out.

**Comment by spislucka** [comment]
I tried installing the browser from the UW script but it didn't work. I downloaded chrome the old fashioned way but chrome now says " The browser is managed by your organization"
Whats up with that?

Edit
also when I try enable Use graphics acceleration when available its grayed out. "This setting is managed by your administrator"    ???

**Reply by memstechtips** [reply]
Hey there, this is due to policies being set in the Google Chrome browser, see here:
![image](https://github.com/user-attachments/assets/a9943f25-d715-4f0b-9106-740e4f2bcad2)

You can delete them all by running this command in PowerShell or Command Prompt as Admin.
```reg delete "HKLM\Software\Policies\Google\Chrome" /f```

Restart your PC once the command has been run and the "This setting is managed by your administrator/organization" will be gone.

I assume the Google Chrome function didn't work when you tried to download in using the UWScript, on which version of Windows were you when you tried running it if I may ask?

**Reply by spislucka** [reply]
> Hey there, this is due to policies being set in the Google Chrome browser, see here: ![image](https://private-user-images.githubusercontent.com/172593213/385042852-a9943f25-d715-4f0b-9106-740e4f2bcad2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEzNTcwMzIsIm5iZiI6MTczMTM1NjczMiwicGF0aCI6Ii8xNzI1OTMyMTMvMzg1MDQyODUyLWE5OTQzZjI1LWQ3MTUtNGYwYi05MTA2LTc0MGU0ZjJiY2FkMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExMVQyMDI1MzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02N2M0MGExZjRiZTNmN2E5N2ZkYmRmNjBmZTYzZmU4MTNlODVjZjNkYmE2NzRkYmMxYTUxZmM3MGU1NTM4YTgxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.SCYAy2LOYok4NkNqUiWL24nyTeFTrOA0kZs0i8yT80M)
> 
> You can delete them all by running this command in PowerShell or Command Prompt as Admin. `reg delete "HKLM\Software\Policies\Google\Chrome" /f`
> 
> Restart your PC once the command has been run and the "This setting is managed by your administrator/organization" will be gone.
> 
> I assume the Google Chrome function didn't work when you tried to download in using the UWScript, on which version of Windows were you when you tried running it if I may ask?

thanks for quick respons, I will try that. Running Win10 22H2 19045.5011

edit2 it worked! Chrome was feeling very slow and laggy, all that is gone now! 
Thanks!

**Reply by horaceBean** [reply]
> Hey there, this is due to policies being set in the Google Chrome browser, see here: ![image](https://private-user-images.githubusercontent.com/172593213/385042852-a9943f25-d715-4f0b-9106-740e4f2bcad2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3MzIyMzYsIm5iZiI6MTczMjczMTkzNiwicGF0aCI6Ii8xNzI1OTMyMTMvMzg1MDQyODUyLWE5OTQzZjI1LWQ3MTUtNGYwYi05MTA2LTc0MGU0ZjJiY2FkMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyN1QxODI1MzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yYWUxMjYyNTY0N2ZhOTJlMTg5ZTE2NGE2YmEzZWEzZjgzMmE0N2RmYTM3MzczZjYzMjZjMTNhMTgzZjZiMDYzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.2JGSCSoZgc46qCGO8fhh2KUhRkDnpnj8MZa84SgAB7k)
> 
> You can delete them all by running this command in PowerShell or Command Prompt as Admin. `reg delete "HKLM\Software\Policies\Google\Chrome" /f`
> 
> Restart your PC once the command has been run and the "This setting is managed by your administrator/organization" will be gone.
> 
> I assume the Google Chrome function didn't work when you tried to download in using the UWScript, on which version of Windows were you when you tried running it if I may ask?

I am currently running Windows 10 Pro. Version 22H2. I'm trying to get rid of  " The browser is managed by your organization". I ran the script above, but I'm getting error: system is unable to find registry key or value. See image 
<img width="566" alt="powershell" src="https://github.com/user-attachments/assets/cd5469c2-b159-41bf-bbca-de2f22ddb063">
What am I doing wrong?

**Comment by Aadi-435** [comment]
![IMG_20241114_224206.jpg](https://github.com/user-attachments/assets/0eb02470-f7de-4b09-b1b1-5c91a5657499)

![IMG_20241114_225828.jpg](https://github.com/user-attachments/assets/1fae317f-bb82-4d20-92a5-ca3ace3a2cf8)

![IMG_20241114_225813.jpg](https://github.com/user-attachments/assets/d8c110f2-b813-4862-b6e6-0ddfccc5b981)


I followed the instructions well but it shows windows installation failed! 

Please help me with this
I even cross checked for some important steps like the file was named as unattended.xml as mentioned 
 
I chose legacy BIOS MODE

**Comment by spislucka** [comment]
Are there any possibility that this mod will fuck with my internet connection? My friend complaining my voice stutters "as if im downloading something very big" when we woip. 
Most new tabs in chrome takes like 5 sec to start loading. 
Speed test is all good. 
wired connection.

This was not a problem before.

**Reply by Miraakz9** [reply]
I've been on v1.0.0 of UnattendedWinstall with Windows 11 for 4 months now and this is the same issue I'm facing, stuttering audio, laggy browser performance (tried every browser, enabled hardware acceleration). Battery icon is also buggy (I'm a laptop user).

Now I was going to try v2.0.0 with windows 10 22H2 that I have downloaded right now but it seems the issues are still there and probably unique to some users too? I absolutely love this project and would like it to be improved but I just want to know why the browser performance is so laggy. It's weird that on my system the gaming performance of Windows 11 is better than Windows 10 but the browser performance is atrocious. Sites with heavy user interfaces like Twitch and Youtube are very slow.

**Comment by hotroorg** [comment]
ERROR: Disables Windows Defender services by default
prompted to enable after Windows installation


>>Windows Defender automatically turns on while installing Windows 11, please guide me to turn off Windows Defender while installing. thank you

**Reply by ghost** [reply]
With VirtualBox it does not happen but same image on real PC will enable Defender during installation.
